src/coreclr/vm/gchelpers.cpp

```diff
@@ -339,12 +402,51 @@ void PublishObjectAndNotify(TObj* &orObject, GC_ALLOC_FLAGS flags)
     {
         ETW::TypeSystemLog::SendObjectAllocatedEvent(orObject);
     }
+
+    if (isSampled && ETW_TRACING_CATEGORY_ENABLED(MICROSOFT_WINDOWS_DOTNETRUNTIME_PROVIDER_DOTNET_Context,
```

Comment:
```suggestion
    if (isSampled)
```
We shouldn't need to check the flags again. If isSampled=true we've already checked.